### PR TITLE
PWGHF: mass fitter for d2h

### PR DESCRIPTION
**Pull request by Mingyu3360715**
first version mass fitter for PWGHF/D2H

### CODE REVIEW

**Comment by vkucera** [comment]
Hi @Mingyu3360715 , thanks for this important development.
At first glance it seems that you are mixing several different naming conventions, especially for the class methods and variables.
Please stick with the official ones.

**Comment by vkucera** [comment]
Hi @Mingyu3360715 , any updates?

**Comment by Mingyu3360715** [comment]
> Hi @Mingyu3360715 , any updates?

Hello Vit. Zhen Zhang and I made some improvements on the fitting performance. And I also prepared a python version of RunMassFitter.C. Tomorrow I will give report about this at the D2H meeting. If there is no big problem, I guess I will push the new version within this week.

**Comment by vkucera** [comment]
> 

Thanks. Will the new version have the names fixed?

**Comment by Mingyu3360715** [comment]
> Thanks. Will the new version have the names fixed?

Yes, Zhen has already fixed it.

**Comment by vkucera** [comment]
When can we expect the additional commit?

**Comment by Mingyu3360715** [comment]
My VPN is out of service recently and I can't get access to Github till now. Today I asked Zhen Zhang to help commit, he will do it later today or tomorrow.




Stark
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "AliceO2Group/O2Physics"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月26日(星期三) 晚上11:49
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [AliceO2Group/O2Physics] PWGHF: mass fitter for d2h (PR #1344)





 
When can we expect the additional commit?

**Comment by vkucera** [comment]
Hi, can you let us know whether you managed to advance with this?

File: PWGHF/D2H/Macros/HFInvMassFitter.cxx [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFInvMassFitter.cxx
+/// \brief HF Invariant Mass Fitter task
```
Comment (vkucera) [inline PWGHF/D2H/Macros/HFInvMassFitter.cxx:13]: Same comment

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/D2H/Macros/runMassFitter.C [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+  }
+
+  // fit histograms
+  Double_t massDplus = TDatabasePDG::Instance()->GetParticle(411)->Mass();
+  Double_t massDs = TDatabasePDG::Instance()->GetParticle(431)->Mass();
+  Double_t massD0 = TDatabasePDG::Instance()->GetParticle(421)->Mass();
+  Double_t massLc = TDatabasePDG::Instance()->GetParticle(4122)->Mass();
+  Double_t dmassDstar = TDatabasePDG::Instance()->GetParticle(413)->Mass() -
+                        TDatabasePDG::Instance()->GetParticle(421)->Mass();
+  Double_t massForFit = -1.;
```
Comment (vkucera) [inline PWGHF/D2H/Macros/runMassFitter.C:374]: There is nothing related to fitting here.

File: PWGHF/D2H/Macros/runMassFitter.C [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+  TH1F* hMassForRefl[nPtBins];
+  TH1F* hMassForSgn[nPtBins];
+
+  Int_t canvasSize[2] = {1920, 1080};
+  if (nPtBins == 1) {
+    canvasSize[0] = 500;
+    canvasSize[1] = 500;
+  }
+
+  Int_t nMaxCanvases = 20; // do not put more than 20 bins per canvas to make them visible
```
Comment (vkucera) [inline PWGHF/D2H/Macros/runMassFitter.C:399]:
```suggestion
  Int_t nCanvasesMax = 20; // do not put more than 20 bins per canvas to make them visible
```

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/D2H/Macros/HFInvMassFitter.cxx [hunk 4]
```diff
@@ -0,0 +1,805 @@
...
+
+/// \file HFInvMassFitter.cxx
+/// \brief HFInvMassFitter class
+///
+/// \author Zhen Zhang <zhenz@cern.ch>
+/// \author Mingyu Zhang <mingyu.zang@cern.ch>
+/// \author Xinye Peng  <xinye.peng@cern.ch>
+/// \author Biao Zhang <biao.zhang@cern.ch>
+///
+#include "RooRealVar.h"
```
Comment (vkucera) [inline PWGHF/D2H/Macros/HFInvMassFitter.cxx:20]:
```suggestion

#include "RooRealVar.h"
```

File: PWGHF/D2H/Macros/HFInvMassFitter.h [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+/// \file HFInvMassFitter.h
+/// \brief HFInvMassFitter class
+///
+/// \author Zhen Zhang <zhenz@cern.ch>
+/// \author Mingyu Zhang <mingyu.zang@cern.ch>
+/// \author Xinye Peng  <xinye.peng@cern.ch>
+/// \author Biao Zhang <biao.zhang@cern.ch>
+///
+#ifndef PWGHF_D2H_MACROS_HFINVMASSFITTER_H_
```
Comment (vkucera) [inline PWGHF/D2H/Macros/HFInvMassFitter.h:20]:
```suggestion

#ifndef PWGHF_D2H_MACROS_HFINVMASSFITTER_H_
```

File: PWGHF/D2H/Macros/runMassFitter.C [hunk 6]
```diff
@@ -0,0 +1,682 @@
...
+
+/// \file runMassFitter.C
+/// \brief HFInvMassFitter class steering macro
+///
+/// \author Zhen Zhang <zhenz@cern.ch>
+/// \author Mingyu Zhang <mingyu.zang@cern.ch>
+/// \author Xinye Peng  <xinye.peng@cern.ch>
+/// \author Biao Zhang <biao.zhang@cern.ch>
+///
+#if !defined(__CINT__) || defined(__CLING__)
```
Comment (vkucera) [inline PWGHF/D2H/Macros/runMassFitter.C:20]:
```suggestion

#if !defined(__CINT__) || defined(__CLING__)
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — APPROVED** [review]
It's not clear to me why the includes are not all in the header file and why `RooWorkspace` is included twice differently, but we can sort this out offline.

**Pull request merged**